be/src/agent/agent_server.cpp

```diff
@@ -67,438 +47,236 @@ AgentServer::AgentServer(ExecEnv* exec_env,
     } else {
-        LOG(INFO) << "make_snapshot success. tablet_id: " << snapshot_request.tablet_id
-                  << " schema_hash: " << snapshot_request.schema_hash << " snapshot_path: " << snapshot_path;
-        return_value.__set_snapshot_path(snapshot_path);
+        LOG(INFO) << "success to make_snapshot. tablet_id=" << snapshot_request.tablet_id
+                  << ", schema_hash=" << snapshot_request.schema_hash
+                  << ", snapshot_path: " << snapshot_path;
+        t_agent_result.__set_snapshot_path(snapshot_path);
     }
 
-    status.__set_error_msgs(error_msgs);
-    status.__set_status_code(status_code);
-    return_value.__set_status(status);
+    ret_st.to_thrift(&t_agent_result.status);
+    t_agent_result.__set_snapshot_version(snapshot_request.preferred_snapshot_version);
     if (snapshot_request.__isset.allow_incremental_clone) {
-        return_value.__set_allow_incremental_clone(snapshot_request.allow_incremental_clone);
+        t_agent_result.__set_allow_incremental_clone(snapshot_request.allow_incremental_clone);
     }
 }
 
-void AgentServer::release_snapshot(TAgentResult& return_value, const std::string& snapshot_path) {
-    vector<string> error_msgs;
-    TStatusCode::type status_code = TStatusCode::OK;
-
-    OLAPStatus release_snapshot_status =
-            SnapshotManager::instance()->release_snapshot(snapshot_path);
-    if (release_snapshot_status != OLAP_SUCCESS) {
-        status_code = TStatusCode::RUNTIME_ERROR;
-        LOG(WARNING) << "release_snapshot failed. snapshot_path: " << snapshot_path << ", status: " << release_snapshot_status;
-        error_msgs.push_back("release_snapshot failed. status: " +
-                             boost::lexical_cast<string>(release_snapshot_status));
+void AgentServer::release_snapshot(TAgentResult& t_agent_result, const std::string& snapshot_path) {
+    Status ret_st;
+    OLAPStatus err_code = SnapshotManager::instance()->release_snapshot(snapshot_path);
+    if (err_code != OLAP_SUCCESS) {
+        LOG(WARNING) << "failt to release_snapshot. snapshot_path: " << snapshot_path
+                     << ", err_code: " << err_code;
+        ret_st = Status::RuntimeError(strings::Substitute(
+                    "fail to release_snapshot. err_code=$0", err_code));
     } else {
-        LOG(INFO) << "release_snapshot success. snapshot_path: " << snapshot_path << ", status: " << release_snapshot_status;
+        LOG(INFO) << "success to release_snapshot. snapshot_path=" << snapshot_path
+                  << ", err_code=" << err_code;
     }
-
-    return_value.status.__set_error_msgs(error_msgs);
-    return_value.status.__set_status_code(status_code);
+    ret_st.to_thrift(&t_agent_result.status);
 }
 
-void AgentServer::publish_cluster_state(TAgentResult& _return, const TAgentPublishRequest& request) {
-    vector<string> error_msgs;
+// TODO(lingbin): always return OK?
+void AgentServer::publish_cluster_state(TAgentResult& t_agent_result,
```

Comment:
The following 4 methods are useless now. I think you can just make it return ERROR, and delete them after.
```
publish_cluster_state
submit_etl_task
get_etl_status
delete_etl_files
```

Reply:
Done.

Reply:
Unfortunately, `Multi Load` will use these `xxx_etl_xxx()` methods, so they will be kept.